From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor@armbian.com>
Date: Tue, 13 Oct 2026 14:58:29 +0200
Subject: [PATCH 12/13] board: bananapi: bpi-w2: load the initrd at 0x30000000

Booted from eMMC, the boot firmware keeps data around 0x40100000 that
the secondary CPUs need on their way to U-Boot. An initrd at
0x40000000 overwrote it, and the CPUs did not come up.

Signed-off-by: Igor Pecovnik <igor@armbian.com>
---
 board/bananapi/bpi-w2/bpi-w2.env | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/board/bananapi/bpi-w2/bpi-w2.env b/board/bananapi/bpi-w2/bpi-w2.env
index 62c22ddd0a..ba24283144 100644
--- a/board/bananapi/bpi-w2/bpi-w2.env
+++ b/board/bananapi/bpi-w2/bpi-w2.env
@@ -1,12 +1,14 @@
 /*
  * Linux reserves most of the low 512 MiB (firmware, heaps, TEE): load
- * everything at 0x1a200000 - 0x1fc00000 or above 0x20000000.
+ * everything at 0x1a200000 - 0x1fc00000 or above 0x20000000. Booted from
+ * eMMC, the boot firmware keeps the secondary CPUs' way to 0x20000 around
+ * 0x40100000: load nothing at 0x40000000 - 0x40200000.
  */
 scriptaddr=0x1fe00000
 pxefile_addr_r=0x1fd00000
 fdt_addr_r=0x20000000
 kernel_addr_r=0x20200000
-kernel_comp_addr_r=0x30000000
+kernel_comp_addr_r=0x38000000
 kernel_comp_size=0x4000000
-ramdisk_addr_r=0x40000000
+ramdisk_addr_r=0x30000000
 fdtfile=realtek/rtd1296-bananapi-w2.dtb
